### PR TITLE
Merge LogStructured in SQLLog

### PR DESCRIPTION
**Pull request by marco6**
This PR merges the functionality from LogStructured into SQLLog and removes LogStructured.

There are many reasons for this change but the most important one is the lack of a real separation between the two structs, making LogStructured a mere translation layer which adds no useful functionality (and actually acts as a limitation, forcing the implementation to be sub-optimal).

The PR is quite big, but each commit takes care of a single aspect of the transition, making it hopefully possible to perform a review.

Wherever made sense the PR also takes the opportunity to make some of the sub-optimal things better both in term of correctness and in terms of memory footprint/speed (for example avoiding some useless copy-paste or some useless memory allocation).

Some part of this PR (namely, the `List`/`Get` query) could be improved/made simpler, however this PR does not attempt to do that, so that it should be compatible with other outstanding PRs.

As a last note, the write path and the indexes *do not change*, so this PR is compatible with all existing versions.

### CODE REVIEW

**Comment by marco6** [comment]
Depends on #193

File: pkg/kine/sqllog/sqllog.go [hunk 1]
```diff
@@ -534,9 +627,67 @@ func (s *SQLLog) notifyWatcherPoll(revision int64) {
...
+	return result, nil
+}
+
+func scanKeyValue(rows *sql.Rows) (*server.KeyValue, error) {
+	kv := &server.KeyValue{}
+	var create, delete bool
+	var prevRevision int64
+	var prevValue []byte
+
+	// FIXME eliminare roba inutile
```
Comment (louiseschmidtgen) [inline pkg/kine/sqllog/sqllog.go:663]: Love the comments haha.

File: pkg/kine/sqllog/sqllog.go [hunk 2]
```diff
@@ -96,6 +98,12 @@ func (s *SQLLog) Start(ctx context.Context) error {
 		return err
 	}
 
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+		s.ttl(ctx)
+	}()
```
Comment (louiseschmidtgen) [inline pkg/kine/sqllog/sqllog.go:107]: Was this not started in the start compaction function? Why move this here?

File: pkg/kine/broadcaster/broadcaster.go [hunk 3]
```diff
@@ -5,21 +5,21 @@ import (
 	"sync"
 )
 
-type ConnectFunc func(ctx context.Context) (chan interface{}, error)
+type ConnectFunc[T any] func(ctx context.Context) (chan T, error)
```
Comment (louiseschmidtgen) [inline pkg/kine/broadcaster/broadcaster.go:8]: Nice!

**Review by louiseschmidtgen — COMMENTED** [review]
Awesome, thanks for splitting this into separate commits this helped tremendously to follow along the changes one-by-one.
A few questions while I was reading through popped up:

File: pkg/kine/sqllog/logstructured.go [hunk 4]
```diff
@@ -127,11 +124,13 @@ func (l *LogStructured) List(ctx context.Context, prefix, startKey string, limit
 
 func (l *LogStructured) ttlEvents(ctx context.Context) chan *server.Event {
 	result := make(chan *server.Event)
-	var shouldClose atomic.Bool
```
[the remark this answers is not in the document]
Comment (marco6) [reply pkg/kine/sqllog/logstructured.go]: Yep.

File: pkg/kine/sqllog/sqllog.go (hunk 1 above)
Comment (marco6) [reply pkg/kine/sqllog/sqllog.go:661]: Sorry, I'll remove this.

**Comment by marco6** [comment]
I rebased on top of `master`, this way we see the benchmark diff which does not include other changes.

**Comment by marco6** [comment]
Dropped dependency from #193

File: pkg/kine/sqllog/sqllog.go (hunk 2 above)
Comment (louiseschmidtgen) [reply pkg/kine/sqllog/sqllog.go:107]: Fair enough, that makes a lot of sense.

**Review by louiseschmidtgen — APPROVED** [review]
LGTM, just removing the comment before merging.

File: pkg/kine/sqllog/sqllog.go (hunk 1 above)
Comment (louiseschmidtgen) [reply pkg/kine/sqllog/sqllog.go]: removed comment

**Pull request merged**